torch/fx/experimental/symbolic_shapes.py

```diff
@@ -992,8 +995,17 @@ def duck_int(self, val):
     # on if the guards in the list evaluated to True or not.  Primarily used by Dynamo,
     # but this is also helpful for manual testing of guards (see
     # evaluate_guards_for_args)
+    #
+    # For convenience in testing, a source is allowed to be a str,
+    # in which case we will assume it is a LocalSource
+    #
+    # simplified lets you omit duck sizing, equality and 0/1 guards.
+    # This is useful for testing when you don't care about the boilerplate
+    # guards, and it may be helpful for user output too (be careful though;
+    # some equality guards are nontrivial!  It would be nice to get simplified
+    # output to print them too)
     def produce_guards(self, placeholders, sources,
```

Comment:
I'm not sure a compilation protecting API should offer a simplified version that actively removes some of the safety. It feels like we should always emit them all, and if you need to remove simplified after, do so with a helper func.

Reply:
Can we cut the difference by making this a private kwarg?

Reply:
making it a private kwarg sounds good to me!